openff/toolkit/utils/rdkit_wrapper.py

```diff
@@ -339,21 +377,38 @@ def from_file_obj(
 
         mols = []
 
+        file_format = normalize_file_format(file_format)
+
         if (file_format == "MOL") or (file_format == "SDF"):
             # TODO: Iterate over all mols in file_data
-            for rdmol in Chem.ForwardSDMolSupplier(file_obj):
-                mol = self.from_rdkit(rdmol, _cls=_cls)
-                mols.append(mol)
+            sdf_supplier = Chem.ForwardSDMolSupplier(
+                file_obj, removeHs=False, sanitize=False, strictParsing=True
+            )
+            mols.extend(
+                self._process_sdf_supplier(
+                    sdf_supplier,
+                    allow_undefined_stereo=allow_undefined_stereo,
+                    _cls=_cls,
+                )
+            )
 
-        if file_format == "SMI":
-            # TODO: Find a cleaner way to parse SMILES lines
-            file_data = file_obj.read()
-            lines = [line.strip() for line in file_data.split("\n")]
-            # remove blank lines
-            lines.remove("")
-            for line in lines:
-                mol = self.from_smiles(line, _cls=_cls)
-                mols.append(mol)
+        elif file_format == "SMI":
+            # There's no good way to create a SmilesMolSuppler from a string
+            # other than to use a temporary file.
+            with tempfile.NamedTemporaryFile(suffix=".smi") as tmpfile:
+                content = file_obj.read()
+                if isinstance(content, str):
+                    # Support the older API, whihc required Unicode strings
+                    tmpfile.write(content.encode("utf8"))
```

Comment:
Nifty! It's cool that you know this.

Reply:
To clarify, that's OpenFF's "older" API. I know this because when I passed in a ByteIO() it failed. There's also a test for this backwards compatibility support:
```
    def test_from_file_obj_smi_supports_stringio(self):
        # Backwards compability. Older versions of OpenFF supported
        # string-based file objects, but not file-based ones.
        with open(file_manager.caffeine_smi) as file_obj:
            mol = self.toolkit_wrapper.from_file_obj(file_obj, "SMI")[0]
        assert mol.name == "CHEMBL113"
```

Grr. I just realized that "file-based" should be "binary-based".  Based on your comment, I see that code comment for the test should instead be in the `rdkit_wrapper` code you highlighted. Just committed and pushed.